### PR TITLE
For #24098: Remove wrapper from tabs tray metrics.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/metrics/Event.kt [hunk 1]
```diff
@@ -203,41 +202,6 @@ sealed class Event {
...
 
-    // Tab tray
-    object TabsTrayOpened : Event()
-    object TabsTrayClosed : Event()
-    data class OpenedExistingTab(val source: String) : Event() {
-        override val extras = mapOf(TabsTray.openedExistingTabKeys.source to source)
-    }
-    data class ClosedExistingTab(val source: String) : Event() {
-        override val extras = mapOf(TabsTray.closedExistingTabKeys.source to source)
-    }
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/fenix/components/metrics/Event.kt:214]: @travis79 Does moving away from the deprecated `record(map) ` to `record(object) ` change something in the way the data is collected. Do we need to notify DS of such changes? Thank you!

Comment (travis79) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Event.kt:214]: Great call to check on this! The data is actually in the same format for the purpose of analysis, so there shouldn't be any need to inform DS about updating away from the deprecated API.

File: app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabsAutoCloseDialogController.kt [hunk 2]
```diff
@@ -25,7 +26,9 @@ class InactiveTabsAutoCloseDialogController(
     fun close() {
         markDialogAsShown()
         refreshInactiveTabsSection()
-        metrics.track(Event.TabsTrayAutoCloseDialogDismissed)
+        org.mozilla.fenix.GleanMetrics.TabsTray.autoCloseSeen.record(NoExtras())
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabsAutoCloseDialogController.kt:29]: nit: use `TabsTrayMetrics`

File: app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt [hunk 3]
```diff
@@ -123,14 +122,14 @@ class DefaultTabsTrayControllerTest {
     fun `GIVEN private mode WHEN handleOpeningNewTab is called THEN Event#NewPrivateTabTapped is added to telemetry`() {
         createController().handleOpeningNewTab(true)
 
-        verify { metrics.track(Event.NewPrivateTabTapped) }
+        assertTrue(TabsTray.newPrivateTabTapped.testHasValue())
```
Comment (rocketsroger) [inline app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt:125]: nit: should we `assertFalse` on `testHasValue` before the controller call?  Apply to other cases as well.

**Review by rocketsroger — APPROVED** [review]
LGTM 🚢 couple small nits.  Thanks!

**Pull request merged**